### PR TITLE
fix(Dockerfile,mc,build.sh): add support for downloading tarballs from minio

### LINKED ISSUE #4: fix(Dockerfile): store mc binary and download on docker build
they don't have 'official' releases now, but it should be good enough to download from dl.minio.io for now

### PR DESCRIPTION
**Pull request by arschles**
Fixes #4 and makes progress on https://github.com/deis/deis/issues/4743 (under `deis-builder` heading).

relies on https://github.com/deis/builder/pull/9

### CODE REVIEW

File: rootfs/builder/build.sh [hunk 1]
```diff
@@ -20,8 +20,8 @@ if ! [[ -z "${TAR_URL}" ]]; then
 			secretKey=`cat /var/run/secrets/object/store/access-secret-key`
 			domain=`echo $TAR_URL | awk -F/ '{print $3}'`
 			echo $keyID $secretKey $domain
-			mc --quiet config host add $domain $keyID $secretKey
-			mc --quiet cp $TAR_URL /tmp/slug.tgz
+			/bin/mc --quiet config host add "http://$domain" $keyID $secretKey
```
Comment (smothiki) [inline rootfs/builder/build.sh]: This wont work if the URL is an aws  S3 based I guess.

Comment (smothiki) [inline rootfs/builder/build.sh]: Is there any specific reason you prefixed http://

File: rootfs/builder/build.sh [hunk 2]
```diff
@@ -199,8 +199,8 @@ if [[ "$slug_file" != "-" ]]; then
 					secretKey=`cat /var/run/secrets/object/store/access-secret-key`
 					domain=`echo $put_url | awk -F/ '{print $3}'`
 					echo $keyID $secretKey $domain
-					mc --quiet config host add $domain $keyID $secretKey
-					mc --quiet cp $slug_file $put_url/
+					/bin/mc --quiet config host add "http://$domain" $keyID $secretKey
```
Comment (bacongobbler) [inline rootfs/builder/build.sh]: /bin is added to $PATH by default so you shouldn't need to give the absolute path here. :)

File: rootfs/builder/build.sh (hunk 1 above)
Comment (arschles) [inline rootfs/builder/build.sh]: why won't it work with s3 based URLs? do they already come with the http:// prefix? the `mc` tool rejects URLs without `http://` prefixed

File: rootfs/builder/build.sh (hunk 2 above)
Comment (arschles) [inline rootfs/builder/build.sh]: yep, you're right, I'm just paranoid so I put it here. will remove

**Pull request merged**